STM/src/main/java/org/jboss/stm/internal/reflect/InvocationHandler.java

```diff
@@ -192,6 +194,18 @@ public InvocationHandler (RecoverableContainer<T> cont, T obj, Uid u)
             }
             else
                 _txObject = new LockManagerProxy<T>(obj, cont);  // recoverable or persistent
+
+            AtomicAction action = new AtomicAction();
+
+            action.begin();
+            int result = _txObject.setlock((_optimistic ? new OptimisticLock(LockMode.WRITE) : new Lock(LockMode.WRITE)), 0, 0);
+
+            if (result != GRANTED)
+                if (txojLogger.logger.isInfoEnabled())
```

Comment:
I wonder if it might be better to throw an exception here like you do on line 415 at the end of the method interceptor:

```java
throw new LockException(Thread.currentThread()+" could not set "+LockMode.stringForm(cachedLock._lockType)+" lock. Got: "+LockResult.stringForm(result));
```